File: Angular/app-museo-ibq/src/app/shared/models/obra.ts
import { Autor } from './autor';
import { Tecnica } from './tecnica';
import { Material } from './material';
import { Imagen } from './imagen';

export interface Obra {
  id: number;
  titulo: string;
  datacion: string | null;
  anio: number | null;
  dimensiones: string | null;
  tipologia: string | null;
  descripcion: string | null;
  marcas_inscripciones: string | null;
  referencias: string | null;
  fecha_ingreso: string | null;
  modo_ingreso: string | null;
  procedencia: string | null;
  estado_conservacion: string | null;
  restauraciones: string | null;
  ubicacion: string | null;
  observaciones: string | null;
  autor: Autor | null;
  tecnica: Tecnica | null;
  materiales: Material[] | null;
  imagenes: Imagen[] | null;
}

export interface ObraRequest {
  titulo: string;
  datacion: string | null;
  anio: number | null;
  dimensiones: string | null;
  tipologia: string | null;
  descripcion: string | null;
  marcas_inscripciones: string | null;
  referencias: string | null;
  fecha_ingreso: string | null;
  modo_ingreso: string | null;
  procedencia: string | null;
  estado_conservacion: string | null;
  restauraciones: string | null;
  ubicacion: string | null;
  observaciones: string | null;
  autorId: number;
  tecnicaId: number;
}

File: Sprinboot/galeiraibq/src/main/java/pelayo/proyecto/galeiraibq/model/Obra.java
package pelayo.proyecto.galeiraibq.model;

import jakarta.persistence.*;
import lombok.Data;
import org.hibernate.annotations.SQLRestriction;

import java.io.Serializable;
import java.math.BigInteger;
import java.util.Date;
import java.util.List;


@Entity
@SQLRestriction("estado_borrado = false")
@Data
@Table(name = "obra")
public class Obra implements Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String titulo;
    private String datacion;
    private Integer anio;
    private String dimensiones;
    private String tipologia;
    private String descripcion;
    private String marcas_inscripciones;
    private String referencias;
    private Date fecha_ingreso;
    private String modo_ingreso;
    private String procedencia;
    private String estado_conservacion;
    private String restauraciones;
    private String ubicacion;
    private String observaciones;
    private Boolean estado_borrado;

    @OneToMany(mappedBy = "obra")
    private List<Imagen> imagenes;
    @OneToMany(mappedBy = "obra")
    private List<ObraMaterial> obraMateriales;

    @ManyToOne
    @JoinColumn(name = "tecnica_id")
    private Tecnica tecnica;

    @ManyToOne
    @JoinColumn(name = "autor_id")
    private Autor autor;


}
